storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/serialization/DepositsFromBlockEventSerializer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.rocksdb.serialization;
+
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.ssz.SSZ;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.pow.event.Deposit;
+import tech.pegasys.teku.pow.event.DepositsFromBlockEvent;
+
+public class DepositsFromBlockEventSerializer implements RocksDbSerializer<DepositsFromBlockEvent> {
+
+  @Override
+  public DepositsFromBlockEvent deserialize(final byte[] data) {
+    return SSZ.decode(
+        Bytes.of(data),
+        reader -> {
+          final UnsignedLong blockNumber = UnsignedLong.fromLongBits(reader.readUInt64());
```

Comment:
If we make the `DepositsFromBlockEvents` implement `SimpleOffsetSerializable`, and add `DepositsFromBlockEvents` to the list of classes parsed for deserialization in `setConstants`, we should be able to use SSZ for serialization and deserialization here: 

- `SimpleOffsetSerializer.serialize(depositsFromBlockEvents)`

- `SimpleOffsetSerializer.deserialize(depositsFromBlockEvents, DepositsFromBlockEvents.class)`

Reply:
Yep, we started there, but it got complicated with the list. in the end @ajsutton said the SimpleOffsetSerializable was probably going to be going away at some point, and we wrote our own serializers...

Reply:
The bigger factor was that we don't want `DepositsFromBlockEvents` to be a subtle but critical piece of database compatibility.  `SimpleOffsetSerializer` will happily just serialise something different and fail to deserialise old versions if there's ever a change to even the order of fields in `DepositsFromBlockEvents` whereas this separate serialiser will maintain the same format (and if fields are added or removed a compiler error will make us deal with them - very clearly in a database related class).

Reply:
That makes sense.